mmpose/datasets/datasets/top_down/topdown_posetrack18_dataset.py

```diff
@@ -186,15 +190,22 @@ def evaluate(self, outputs, res_folder, metric='mAP', **kwargs):
                     kpt_score = kpt_score / valid_num
                 # rescoring
                 n_p['score'] = kpt_score * box_score
+
+            nms = soft_oks_nms if self.soft_nms else oks_nms
+            keep = nms(list(img_kpts), oks_thr, sigmas=self.sigmas)
+
             if self.use_nms:
                 nms = soft_oks_nms if self.soft_nms else oks_nms
                 keep = nms(list(img_kpts), oks_thr, sigmas=self.sigmas)
-                valid_kpts[image_id].append(
-                    [img_kpts[_keep] for _keep in keep])
+                if len(keep) == 0:
```

Comment:
why check this?

If keep is empty, then img_kpts is also empty. Then what's the difference between your `if` and `else`?